vendor/github.com/gravitational/teleport/lib/utils/utils.go

```diff
@@ -361,18 +361,23 @@ func (p *PortList) PopIntSlice(num int) []int {
 	return ports
 }
 
-// PortStartingNumber is a starting port number for tests
-const PortStartingNumber = 20000
-
-// GetFreeTCPPorts returns n ports starting from port 20000.
-func GetFreeTCPPorts(n int, offset ...int) (PortList, error) {
+// GetFreeTCPPorts returns n free ports (which are suggested by the kernel)
+func GetFreeTCPPorts(n int) (PortList, error) {
 	list := make(PortList, 0, n)
-	start := PortStartingNumber
-	if len(offset) != 0 {
-		start = offset[0]
-	}
-	for i := start; i < start+n; i++ {
-		list = append(list, strconv.Itoa(i))
+	for i := 0; i < n; i++ {
+		addr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		listen, err := net.ListenTCP("tcp", addr)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		defer listen.Close()
+
+		port := strconv.Itoa(listen.Addr().(*net.TCPAddr).Port)
+		list = append(list, port)
```

Comment:
Make a new function outside of `vendor/`. Otherwise, next time someone updates deps this will get overwritten.

Reply:
Added to `utils` in `teleport-plugins` and renamed to `GetFreeTCPPortsForTests`.